libs/s25main/network/GameServer.cpp

```diff
@@ -526,6 +526,11 @@ bool GameServer::assignPlayersOfRandomTeams(std::vector<JoinPlayerInfo>& playerI
     return playerWasAssigned;
 }
 
+void GameServer::SetNumPlayers(unsigned num)
+{
+    playerInfos.resize(num);
```

Comment:
We should add a check `(state == ServerState::Config)` before doing this. Maybe:
```suggestion
    if(state != ServerState::Config)
        throw std::logic_error("Changing the number of players must only be done during game configuration");
    if(num < 1 || num > MAX_PLAYERS)
        throw std::invalid_argument("Invalid number of players");
    playerInfos.resize(num);
```

Not sure about the 2nd check as the value is likely from a lua script. So maybe rather return false here and abort the game in the caller.